corehq/apps/hqwebapp/tests/test_forms.py

```diff
@@ -0,0 +1,82 @@
+from unittest.mock import patch, MagicMock, ANY
+from django.test import RequestFactory, SimpleTestCase
+from django.core.exceptions import ValidationError
+from corehq.apps.users.models import WebUser
+
+from ..forms import HQAuthenticationTokenForm, HQBackupTokenForm
+
+
+class HQAuthenticationTokenFormTests(SimpleTestCase):
+    def setUp(self):
+        self.factory = RequestFactory()
+        self.setUp_mocks()
+
+    def setUp_mocks(self):
+        user_patcher = patch('corehq.apps.hqwebapp.forms.CouchUser.get_by_username',
+            new=MagicMock(side_effect=lambda x: self.user))
```

Comment:
It looks like `self.user` is initialized in `begin_login_attempt`, which is called in test methods (not part of test setup). Two ideas:

1. What do you think of adding a `tearDown` method to set `del self.user` to ensure the user from one test does not bleed over into another.
2. Alternately, maybe move `self.user = WebUser(username='test_user')` into this `setUp` method?

Reply:
I don't think I'm following this. Is the concern that the code is confusing, and/or that `self.user` is never deleted? For the latter, `self.user` never gets saved to the database, so it doesn't seem like it needs to be deleted?  For the former, I wanted `test_failed_authentication_sends_fully_formed_signal` to make sense. It needs to verify the credentials and the request are as expected.  I feel like asserting against information only in the setup make tests difficult to understand.

Reply:
Both issues you raise are a concern. It's confusing, and if a test is refactored or a new test is added such that it neglects to call `begin_login_attempt()` it will use `self.user` from the previous test.

Another possibly cleaner option would be to move `user_patcher` initialization into `begin_login_attempt()` and have the user object scope be limited to that method (`self.user` need not be assigned).

Reply:
> if a test is refactored or a new test is added such that it neglects to call `begin_login_attempt()` it will use `self.user` from the previous test.

For the public record, I was wrong about this. It is not a concern.